### ISSUE TITLE
Unsupervised Reinforcement Learning Baseline Reproduction

### ISSUE THREAD

**Issue by Veldrovive** [andykr1k/ECE567_Final_Project#3]
Link to baselines implementation
[https://github.com/facebookresearch/controllable_agent](https://github.com/facebookresearch/controllable_agent)

Goal is to reproduce baselines from the papers linked in the repo.

**Comment by skyalter** [comment]
## Training Command
```bash
python -m url_benchmark.pretrain agent=<agent> task=<domain>_<task> goal_space=<goal_space> use_tb=1 use_hiplog=1
```
Default `reward_free=True`. 

The agent is trained without any reward function and can zero-shot optimize arbitrary rewards after training.

---

### Agent

Refer to [`url_benchmark/agent/`](https://github.com/facebookresearch/controllable_agent/tree/main/url_benchmark/agent)

1. `agent=fb_ddpg` — [Forward Backward Representation](https://arxiv.org/pdf/2103.07945)
2. `agent=diayn` — [Diversity is All You Need](https://arxiv.org/pdf/1802.06070)

---

### Task

Format: `task=<domain>_<task>`. 

Sources: 

1. [`dm_control suite`](https://github.com/google-deepmind/dm_control/tree/main/dm_control/suite) via `@SUITE.add()`
2. [`custom_dmc_tasks/`](https://github.com/facebookresearch/controllable_agent/tree/main/url_benchmark/custom_dmc_tasks) via `@SUITE.add()`

- **walker**: 1. `stand`, `walk`, `run` 2. `flip`
- **cheetah**: 1. `run` 2. `walk`, `walk_backward`, `run_backward`, `flip`, `flip_backward`
- **hopper**: 1. `stand`, `hop` 2. `hop_backward`, `flip`, `flip_backward`
- **quadruped**: 1. `walk`, `run`, `escape`, `fetch` 2. `stand`, `jump`, `roll`, `roll_fast`

---

### Goal Space

Refer to [`url_benchmark/goals.py`](https://github.com/facebookresearch/controllable_agent/blob/main/url_benchmark/goals.py) via `@goal_spaces("domain")`

Determines which physical variables are fed into the Backward network. Must match the domain. Cheetah and Hopper have no predefined goal space.

- **walker**: `simplified_walker` (3d), `walker_pos_speed` (4d), `walker_pos_speed_z` (6d)
- **quadruped**: `simplified_quadruped` (2d), `quad_pos_speed` (7d)
- **cheetah**: None
- **hopper**: None

---

### Other Options

Defined in [`pretrain.py`](https://github.com/facebookresearch/controllable_agent/blob/main/url_benchmark/pretrain.py). View all with:
```bash
python -m url_benchmark.pretrain agent=fb_ddpg --cfg job
```

**Comment by Infinite-Echo** [comment]
https://wandb.ai/cfuel-umich-uom/controllable_agent/workspace?nw=nwusercfuelumich

**Comment by Infinite-Echo** [comment]
Not sure if these are actually training properly. If anyone wants to take a look, I'd appreciate it. If you can't view the link let me know.

**Comment by andykr1k** [comment]
It would be great if you could fill out the results section in the overleaf showing the results you got vs the published results.

**Comment by Infinite-Echo** [comment]
I thought someone else was assigned to writing the report. I can do it but it's BS because the envs given to us aren't used in forward backward so there is no comparison and DIAYN doesn't work so I really think I'm not understanding the task properly but no one responded to me over the weekend so who cares.

**Comment by andykr1k** [comment]
Other people are writing the environment and baseline methods. All we need from you is the results subsection for the reproducibility section since you have all the data. I am trying to get others to respond but I'm not getting any hits.

**Comment by Infinite-Echo** [comment]
Reference links for self

FB_DDPG Reward Panel: https://wandb.ai/cfuel-umich-uom/controllable_agent/workspace?nw=nwusercfuelumich&panelDisplayName=eval%2Fepisode_reward&panelSectionName=eval

DIAYN Reward Panel: https://wandb.ai/cfuel-umich-uom/controllable_agent/workspace?nw=nwusercfuelumich&panelDisplayName=eval%2Fepisode_reward&panelSectionName=eval

**Comment by Infinite-Echo** [comment]
Results and reproducibility sections are filled out for Project 1. DIAYN was not reproducible at all. FB-DDPG was mostly reproducible but not directly comparable since the original paper does not use the benchmark's environments.

**Comment by skyalter** [comment]
Our FB-DDPG Walker results are expected — others have reported the same issue when running without a goal space (see https://github.com/facebookresearch/controllable_agent/issues/6).

For DIAYN, I think the poor results are inherent to the method. It only learns discrete skills by maximizing distinguishability, but “distinguishable” is not “useful.” The agent may learn 50 different ways of falling over, none resembling walking. There’s no mechanism guiding skills toward useful behaviors, so increasing skill dimension doesn’t help.

By the way, I’m unable to access those wandb.ai links. My email is sunky@umich.edu or sunkuanyou@gmail.com.
